src/Arcus.Templates.Tests.Integration/Fixture/WebApiProject.cs

```diff
@@ -0,0 +1,98 @@
+        /// <param name="configuration">The configuration to control the hosting of the to-be-created project.</param>
+        /// <param name="outputWriter">The output logger to add telemetry information during the creation and startup process.</param>
+        /// <returns>
+        ///     A web API project with a full set of endpoint services to interact with the API.
+        /// </returns>
+        public static async Task<WebApiProject> StartNewAsync(TestConfig configuration, ITestOutputHelper outputWriter)
+        {
+            Guard.NotNull(configuration, nameof(configuration), "Cannot create a web API project from the template without a test configuration");
+            Guard.NotNull(outputWriter, nameof(outputWriter), "Cannot create web API services without a test output logger");
+
+            return await StartNewAsync(configuration, WebApiProjectOptions.Empty, outputWriter);
+        }
+
+        /// <summary>
+        /// Starts a newly created project from the web API project template with a given set of <paramref name="projectOptions"/>.
+        /// </summary>
+        /// <param name="configuration">The configuration to control the hosting of the to-be-created project.</param>
+        /// <param name="projectOptions">The project options to control the functionality of the to-be-created project from this template.</param>
+        /// <param name="outputWriter">The output logger to add telemetry information during the creation and startup process.</param>
+        /// <returns>
+        ///     A web API project with a full set of endpoint services to interact with the API.
+        /// </returns>
+        public static async Task<WebApiProject> StartNewAsync(TestConfig configuration, WebApiProjectOptions projectOptions, ITestOutputHelper outputWriter)
+        {
+            Guard.NotNull(configuration, nameof(configuration), "Cannot create a web API project from the template without a test configuration");
+            Guard.NotNull(projectOptions, nameof(projectOptions), "Cannot create a web API project without a set of project argument options");
+            Guard.NotNull(outputWriter, nameof(outputWriter), "Cannot create web API services without a test output logger");
+
+            DirectoryInfo templateDirectory = configuration.GetWebApiProjectDirectory();
+            Uri baseUrl = configuration.CreateWebAPiBaseUrl();
+            var project = new WebApiProject(baseUrl, templateDirectory, outputWriter);
+            project.CreateNewProject(projectOptions);
+            
+            project.Run(configuration.BuildConfiguration, $"--ARCUS_HTTP_PORT {baseUrl.Port}");
+            await WaitUntilWebProjectIsAvailable(baseUrl.Port);
+
+            return project;
+        }
+
+        private static async Task WaitUntilWebProjectIsAvailable(int httpPort)
+        {
+            var waitAndRetryForeverAsync =
+                Policy.HandleResult((HttpStatusCode code) => code == HttpStatusCode.NotFound)
```

Comment:
Is this enough if the test is enabling Shared access keys?

Reply:
It's actually a kind of fail-safe so we start the test cases only when we have a running instance.
Without knowing exactly what endpoints the API has; we can try to call an enpoint that we could guess doesn't exists and try that until we don't have a connection failure but a `NotFound`.

We could also call the health endpoint in the same way but trying until we get any response of course.

Reply:
That's a good point, but if we have security enabled, will we ever get NotFound rather than Unauthorized?

Reply:
1. If you mean using the current approach, we'll always end up with a `NotFound` since the endpoint doesn't exists
2. If you mean using the health, than yes, we'ld have to check if we get _any_ response instead of looking for the right status code.

Reply:
Resolving, looks ok. Health would be nice but too much churn given auth can be different.